### PR TITLE
passing spec, new endpoint, service layer

### PR DESCRIPTION
**Pull request by emmajhyde**

### CODE REVIEW

File: app/models/station.rb [hunk 1]
```diff
@@ -2,4 +2,5 @@
 
 class Station < ApplicationRecord
   has_many :favorites
+  has_many :users, through: :favorites
```
Comment (emmajhyde) [inline app/models/station.rb]: didn't know why this wasn't reflected on both sides, so put it back

File: app/services/mbta_service.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    favorite = Favorite.find(favorite_id)
+    station = Station.find(favorite.station_id)
+    response = HTTP.get(resolve_predictions_url(station.mbta_id))
+    json = JSON.parse(response.body)
+    # whatever the structure is...
+    json['data'].map do |prediction|
+      # this whole hash is assigned to direction_and_time,
+      # regardless of if it is arriving or departing.
+
+      direction_and_time = if prediction['attributes']['arrival_time']
```
Comment (emmajhyde) [inline app/services/mbta_service.rb]: Needs to be changed re: some having both

Comment (emmajhyde) [reply app/services/mbta_service.rb]: otherwise 100% functional

File: spec/services/mbta_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
...0","direction_id":0,"schedule_relationship":null,"status":null,"stop_sequence":130},"id":"prediction-43207011-70085-130","relationships":{"route":{"data":{"id":"Red","type":"route"}},"stop":{"data":{"id":"70085","type":"stop"}},"trip":{"data":{"id":"43207011","type":"trip"}},"vehicle":{"data":{"id":"R-5462DE37","type":"vehicle"}}},"type":"prediction"},{"attributes":{"arrival_time":"2020-02-10T23:40:15-05:00","departure_time":"2020-02-10T23:41:02-05:00","direction_id":1,"schedule_relationship":null,"status":null,"stop_sequence":90},"id":"prediction-43207002-70086-90","relationships":{"route":{"data":{"id":"Red","type":"route"}},"stop":{"data":{"id":"70086","type":"stop"}},"trip":{"data":{"id":"43207002","type":"trip"}},"vehicle":{"data":{"id":"R-5462DFD2","type":"vehicle"}}},"type":"prediction"}],"jsonapi":{"version":"1.0"},"links":{"first":"https://api-v3.mbta.com/predictions?filter[stop]=place-jfk&page[limit]=2&page[offset]=0&sort=departure_time","last":"https://api-v3.mbta.com/predictions?filter[stop]=place-jfk&page[limit]=2&page[offset]=24&sort=departure_time","next":"https://api-v3.mbta.com/predictions?filter[stop]=place-jfk&page[limit]=2&page[offset]=2&sort=departure_time"}}')
```
Comment (emmajhyde) [inline spec/services/mbta_service_spec.rb]: response mock, when i call .body on a double (think body double) it returns wahtever i put at the body attribute

File: spec/services/mbta_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  describe "GET /predictions" do
+    it "sure does do it" do
+      allow(HTTP).to receive(:get).and_return(response_from_mbta)
```
Comment (emmajhyde) [inline spec/services/mbta_service_spec.rb]: prevents HTTP from actually making the call, we "override'" its response with the mock

File: db/migrate/20200211040737_rename_mbta_stop_id_just_mbtaid.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class RenameMbtaStopIdJustMbtaid < ActiveRecord::Migration[5.2]
+  def change
+    remove_column :stations, :mbta_stop_id, :string
+    add_column :stations, :mbta_id, :string
+  end
+end
+e
```
Comment (emmajhyde) [inline db/migrate/20200211040737_rename_mbta_stop_id_just_mbtaid.rb]: fck

File: spec/services/mbta_service_spec.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      predictions = MBTAService.get_predictions(favorite.id)
+      expect(predictions[0][:description]).to eq 'Napsack Farts & The China Girls'
+    end
+  end
+end
```
Comment (emmajhyde) [inline spec/services/mbta_service_spec.rb]: upz

File: app/controllers/favorites_controller.rb [hunk 7]
```diff
@@ -38,6 +38,12 @@ def destroy
     @favorite.destroy
   end
 
+  # GET /predictions?id
+  def predictions
+    @favorite = Favorite.find(params[:id])
+    MBTAService.get_predictions(@favorite.id)
```
Comment (emmajhyde) [inline app/controllers/favorites_controller.rb]: untested, but pretty short. i would be surprised if this didnt work

File: spec/services/mbta_service_spec.rb (hunk 4 above)
Comment (emmajhyde) [reply spec/services/mbta_service_spec.rb]: allow ALWAYS should already have been passed before the call is made

File: app/models/station.rb (hunk 1 above)
Comment (derekmurphy1993) [reply app/models/station.rb]: The reasoning was that a station doesnt have ownership over a user, so it wasnt really doing anything

Comment (emmajhyde) [reply app/models/station.rb]: 😼

Comment (emmajhyde) [reply app/models/station.rb]: VBD format aside, it doesnt hurt and now given a station object you could theoretically get all of the stations they liked. valuable data, i need to know what ur fav station is

**Pull request merged**